### ISSUE TITLE
[init-declarations] It report `Variable 'b' should be initialized on declaration.` when I declare nested namespace in a `index.d.ts` file

### ISSUE THREAD

**Issue by drylint** [typescript-eslint/typescript-eslint#4392]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "init-declarations": [0],
    "@typescript-eslint/init-declarations": [2, "always"],
  }
}
```

```TS
// index.d.ts

// eg.1
declare namespace obj {
  const a: number
}

// eg.2
declare namespace obj1 {
  declare const a: number
  namespace obj1_1 {
    const b: string // Error info: Variable 'b' should be initialized on declaration.
  }
}

// eg.3
declare namespace obj2 {
  const a: number
  namespace obj2_2 {
    declare const b: string
  }
}


```

**Expected Result**

Expected that don't report error with `const b: string` in `eg.2`

**Actual Result**

`eg.1` and `eg.3` is OK, but `eg.2` was reported `Variable 'b' should be initialized on declaration.`

![image](https://user-images.githubusercontent.com/80906525/148362568-9d44b6a7-5ae5-4df5-9d54-78d613f92cf3.png)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.8.1` |
| `@typescript-eslint/parser`        | `5.8.1` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.5.0` |
| `node`                             | `16.13.1` |

**Comment by lonyele** [comment]
Just curious. Doesn't any `declare` under the `declare namespace obj { ... }` show some error like `A 'declare' modifier cannot be used in an already ambient context.`?

https://www.typescriptlang.org/play?#code/LAKA9GAECmDmB0BGUATaBjANgQwE7UgDtsBbaAZwAdt0CB7AIwCtIBvUSSdOw8gF0jYAXEQCuJBtFygAvqFAQYCAEyoMOfEVIVqtSIyaI2HSGix4C3XgOFiJUk8TJUa9ZogD6R9iE6cr-JAMIvy4AJaEsJCKAKK4uHS4kBEAZnQiAGp4YdgMmAQA5AwFkOQAFnSimChBBBFhfDmYYQBe0DU8pup42I088CZyIEMKUHDwAMxq5ppOOq76zMrGvlw8gbaE4pLSq3MuegbKHss+fl0zlusCwaV84ZGDskA

**Comment by JoshuaKGoldberg** [comment]
Yup. In theory this is why it's generally preferable to post playground links like that one 👆 instead of screenshots.
